### PR TITLE
Fix benchmarks of saft-registry

### LINKED ISSUE #317: Fix benchmarks for pallet asset-index and pallet saft-registry
## Issue summary

```
 𝝺 ./target/release/pint benchmark -p pallet_asset_index -e "deposit" --chain pint-local                                                                                                                                                                                        
2021-09-01 19:44:50 assembling new collators for new session 0 at #0                                                                                                                                                                                                            
2021-09-01 19:44:50 assembling new collators for new session 1 at #0                                                                                                                                                                                                            
Error: Input("\n* Pallet: AssetIndex\n* Benchmark: deposit\n* Components: []\n* Verify: false\n* Error message: An overflow would occur")                                                                                                                                       
 𝝺 ./target/release/pint benchmark -p pallet_saft_registry -e "add_saft" --chain pint-local                                                                                                                                                                                     
2021-09-01 19:47:18 assembling new collators for new session 0 at #0                                                                                                                                                                                                            
2021-09-01 19:47:18 assembling new collators for new session 1 at #0                                                                                                                                                                                                            
Error: Input("\n* Pallet: SaftRegistry\n* Benchmark: add_saft\n* Components: []\n* Verify: false\n* Error message: Bad origin") 
```

- Benchmarks in pallet `asset-index` has  `Overflow` problem
- Benchmarks in pallet `asset-index` has `BadOrigin` problem
  - seems also has the `overflow` problem


---

- [ ] Check the genesis balances and assets to debug this issue


## Other information and links

- #266 


<!-- Thank you 🙏 -->

### LINKED ISSUE #325: Fix the benchmarks of pallet saft-registry
`Self::nav()` broken, because no available assets yet, 

For could not `add_saft`, the reason is like the issue in https://github.com/ChainSafe/PINT/blob/810ea668ab826e0470de8c791ef83c41bf939f33/pallets/asset-index/src/tests.rs#L115, add another liquid asset will be able to fix it

_Originally posted by @clearloop in https://github.com/ChainSafe/PINT/issues/317#issuecomment-911320472_

Might need to add something like `create_feed` in #300 provided

### PR DESCRIPTION
**Pull request by clearloop**
## Changes

- Based on #323
- Adds trait `AssetsRecorderBenchmarks`
- Fix all benchmarks

## Tests

```
cargo b --release --features runtime-benchmarks
```

## Issues

- Closes #317
- Closes #325

### CODE REVIEW

File: pallets/asset-index/src/benchmarking.rs [hunk 1]
```diff
@@ -110,39 +99,57 @@ benchmarks! {
...
+	// 	));
+	//
+	// 	// create price feed
+	// 	T::PriceFeedBenchmarks::create_feed(origin_account_id.clone(), asset_id).unwrap();
+	//
+	// 	// construct call
+	// 	let call = Call::<T>::remove_asset(asset_id, units, Some(receiver));
+	// }: { call.dispatch_bypass_filter(origin.clone())? } verify {
+	// 	assert_eq!(T::IndexToken::total_balance(&origin_account_id), 0u32.into());
+	// }
```
Comment (mattsse) [inline pallets/asset-index/src/benchmarking.rs:152]: is this blocked by something?

File: pallets/asset-index/src/lib.rs [hunk 2]
```diff
@@ -972,16 +974,16 @@ pub mod pallet {
 				Ok(())
 			})?;
 
+			// mint the given units of the SAFT asset into the treasury's account
+			T::Currency::deposit(asset_id, &Self::treasury_account(), units)?;
+
```
Comment (mattsse) [inline pallets/asset-index/src/lib.rs:979]: this must happen after NAV calculation because this affects NAV.

**Review by mattsse — CHANGES_REQUESTED** [review]

File: pallets/asset-index/src/benchmarking.rs (hunk 1 above)
Comment (clearloop) [reply pallets/asset-index/src/benchmarking.rs:152]: This requires calls in `remote-asset-manager`, got `InvalidDest` error of xcm

File: pallets/asset-index/src/lib.rs (hunk 2 above)
Comment (clearloop) [reply pallets/asset-index/src/lib.rs:979]: Great catch! moved for debugging the nav before

File: pallets/asset-index/src/benchmarking.rs (hunk 1 above)
Comment (mattsse) [reply pallets/asset-index/src/benchmarking.rs:152]: ah yes, you're right, the XCM dependent calls can't be benchmarked like that. voting to even remove them.

**Review by mattsse — APPROVED** [review]

**Pull request merged**